### PR TITLE
show user note on waypoint popup (fix #8290)

### LINKED ISSUE #8290: The waypoint note is not shown in the map popup.
**Describe the bug:**
If you create a custom waypoint, your waypoint note will not appear in the map pop-up.

**To Reproduce:**
1. Go to  the cache detail page
2. Create a new waypoint with a note
3. Go to map and click on the waypoint marker
4. The waypoint note is not shown

**Actual behavior/state after performing these steps:**
The own waypoint note is not shown

**Expected behavior/state after performing these steps:**
Your own waypoint note should be shown.

**Version of c:geo used:**
2020.04.25

**Additional context:**
There are two notes for cache waypoints, one from the CO (not editable) and the "Own note".
The CO's waypoint note is displayed, but not your own.
Custom waypoints have no note from the CO. It would be nice if this would also be shown in the Map popup,

### PR DESCRIPTION
**Pull request by moving-bits**
Shows the user note on waypoint popup:

image

If user note is empty, no user note field will be shown:

image

### CODE REVIEW

File: main/src/cgeo/geocaching/WaypointPopupFragment.java [hunk 1]
```diff
@@ -100,7 +100,14 @@ protected void init() {
             //Waypoint geocode
             details.add(R.string.cache_geocode, waypoint.getPrefix() + waypoint.getGeocode().substring(2));
             waypointDistance = details.addDistance(waypoint, waypointDistance);
-            details.addHtml(R.string.waypoint_note, waypoint.getNote(), waypoint.getGeocode());
+            final String note = waypoint.getNote();
+            if (null != note && !note.isEmpty()) {
```
Comment (UniQP) [inline main/src/cgeo/geocaching/WaypointPopupFragment.java:104]:
```suggestion
            if (StringUtils.isNotBlank(note)) {
```

File: main/src/cgeo/geocaching/WaypointPopupFragment.java [hunk 2]
```diff
@@ -100,7 +100,14 @@ protected void init() {
...
+                details.addHtml(R.string.waypoint_note, note, waypoint.getGeocode());
+            }
+            final String userNote = waypoint.getUserNote();
+            if (null != userNote && !userNote.isEmpty()) {
```
Comment (UniQP) [inline main/src/cgeo/geocaching/WaypointPopupFragment.java:108]:
```suggestion
            if (StringUtils.isNotBlank(userNote)) {
```

File: main/src/cgeo/geocaching/WaypointPopupFragment.java (hunk 1 above)
Comment (moving-bits) [reply main/src/cgeo/geocaching/WaypointPopupFragment.java]: fixed

File: main/src/cgeo/geocaching/WaypointPopupFragment.java (hunk 2 above)
Comment (moving-bits) [reply main/src/cgeo/geocaching/WaypointPopupFragment.java]: fixed

**Pull request merged**